src/collection.cpp

```diff
@@ -72,11 +72,36 @@ uint32_t Collection::get_next_seq_id() {
     return next_seq_id++;
 }
 
-Option<bool> Collection::add_reference_helper_fields(nlohmann::json& document) {
+Option<bool> single_value_filter_query(nlohmann::json& document, const std::string& field_name,
+                                       const std::string& ref_field_type, std::string& filter_query) {
+    auto const& value = document[field_name];
```

Comment:
Is `field_name` guaranteed to exist in document? (e.g. what if field is optional)

Reply:
Yes, `field_name` is guaranteed to exist. `single_value_filter_query` is only called in `add_reference_helper_fields` and we have these checks before calling `single_value_filter_query `.
* [Check 1](https://github.com/typesense/typesense/blob/de036ba406a2e2ceeeadf65b1c19438437132f58/src/collection.cpp#L105-L107)
* [Check 2](https://github.com/typesense/typesense/blob/de036ba406a2e2ceeeadf65b1c19438437132f58/src/collection.cpp#L217-L219)